gateway/resolver/subscription.go

```diff
@@ -24,6 +24,14 @@ var (
 	ErrFailedToCastEventObjectToUnstructured = fmt.Errorf("failed to cast event object to unstructured")
 )
 
+type SubscriptionError struct {
+	Message string
+}
+
+func (e *SubscriptionError) Error() string {
+	return e.Message
+}
+
 func (r *Service) SubscribeItem(gvk schema.GroupVersionKind, scope v1.ResourceScope) graphql.FieldResolveFn {
 	return func(p graphql.ResolveParams) (interface{}, error) {
 		resultChannel := make(chan interface{})
```

Comment:
Do you ever close the channel? Is there maybe a context it could be bound to? If it's never closed it could lead to resource leakage in the long run.